### ISSUE TITLE
Android:You attempted to set the key `navigatorEventHandler` with the value `undefined` on an object that is meant to be immutable and has been frozen.

### ISSUE THREAD

**Issue by nnanna3wp** [wix/react-native-navigation#1240]
### Issue Description

When trying to add right button to nav menu, I get this error.

You attempted to set the key `navigatorEventHandler` with the value `undefined` on an object that is meant to be immutable and has been frozen.(https://github.com/wix/react-native-navigation/issues/197)]

Code Snipset:

static navigatorButtons = {
    rightButtons: [
      {
        title: '',
        id: 'menu',
        testID: 'e2e_rules', 
        disabled: false, 
        disableIconTint: true,
        showAsAction: 'default', 
        icon: require('../img/menu_button.png')
      },
    ]
  };

constructor(props) {
    super(props);
    this.props.navigator.setOnNavigatorEvent(this.onNavigatorEvent.bind(this));
  }

onNavigatorEvent(event) {
    if (event.id === 'menu') {
      Alert.alert('NavBar', 'Edit button pressed');
    }
  }
![photo4947207399463298990](https://cloud.githubusercontent.com/assets/25151850/25961398/0e1a9d1c-3672-11e7-87ef-a3457f913532.jpg)

---
### Environment
* React Native Navigation version: 1.1.69
* React Native version: 0.44.0
* Platform(s) (iOS, Android, or both?): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Samsung 6.0.1 S5 Debug

**Comment by lekenny** [comment]
how did you fix this

**Comment by saudpunjwani101** [comment]
how did you fix?

**Comment by nnanna3wp** [comment]
Remove icon

**Comment by paalex** [comment]
having exactly the same issue, removing the icon doesn't help (error occurs regardless whether there's a button or not).
Anyone managed to solve this?

RN0.49.3

**Comment by paulbevis** [comment]
I recently had the same issue.  I had an entry screen, which called a second screen via 'this.navigator.push', and this second screen had the error described above.  I noticed that the constructor of this particular screen was being called twice (not sure if this had anything to do with the issue)

When I replaced this 'push' call with 'Navigation.startSingleScreenApp' instead, the error no longer occurred.

Although, this doesn't seem to the correct way to chain screens, i couldn't get around the error when I just pushed another screen on to the stack that required navigation event handling...

RN: 0.49.3
RNN: 1.1.246
Platform: IOS(simulator)
